app/scopes/search_helper.ts

```diff
@@ -101,15 +97,25 @@ export function handleSearchQuery<T extends LucidModel>(): QueryScope<
       const [column, value] = entry;
       if (Array.isArray(value)) {
         query = handleArray(query, column, value);
-      } else if (typeof value === "object") {
-        query = handleFromTo(query, column, value);
       } else {
-        query = handleDirectValue(query, column, value);
+        const range = checkForRangeSearch(value);
+        query =
+          range !== undefined
+            ? handleFromTo(query, column, range)
+            : handleDirectValue(query, column, value);
       }
     }
   });
 }
 
+function checkForRangeSearch(value: string): FromTo | undefined {
+  try {
+    return JSON.parse(value) as FromTo;
+  } catch {
+    return undefined;
+  }
+}
```

Comment:
this check is actually insufficient - remember that `JSON.parse` can parse more than just object, including numbers.
this check will actually treat any number as a range query, thus breaking search by exact value for numbers.